MoppApp/MoppApp/CryptoContainerViewController.swift

```diff
@@ -130,11 +133,32 @@ extension CryptoContainerViewController : ContainerViewControllerDelegate {
     }
     
     func openContainer(afterSignatureCreated: Bool = false) {
+        
         if state != .loading { return }
         if container == nil {
-            let containerFilename = (containerPath as NSString).lastPathComponent
-            container = CryptoContainer.init(filename: containerFilename as NSString, filePath: containerPath as NSString)
-            self.reloadData()
+            let filePath = containerPath as NSString
+            let container = CryptoContainer(filename: filePath.lastPathComponent as NSString, filePath: filePath)
+            MoppLibCryptoActions.sharedInstance().parseCdocInfo(
+                filePath as String?,
+                success: {(_ cdocInfo: CdocInfo?) -> Void in
+                    guard let strongCdocInfo = cdocInfo else { return }
+                    
+                    container.addressees = strongCdocInfo.addressees
+                    container.dataFiles = strongCdocInfo.dataFiles
+                    self.containerPath = filePath as String?
```

Comment:
`weak self` should be used here. But if there was not possible to use it, then this code is OK.